### PR TITLE
devtool: allow fetching private git repos

### PR DESCRIPTION
**Pull request by luminitavoicu**
Signed-off-by: Luminita Voicu <lumivo@amazon.com>

# Reason for This PR

Fixes https://github.com/firecracker-microvm/firecracker/issues/2104

TL;DR: `devtool` runs the `cargo build` command from inside the container. When working with dependencies on libraries which reside inside private git repositories, we need a way to perform git authentication for fetching those dependencies.

## Description of Changes

Added `devtool` build `--ssh-keys` flag to provide a way to specify the paths to the public and private SSH keys on the host. These will be mapped to the guest and used for authentication when fetching from private repositories.

This is how the added functionality was tested:
- Mirrored one of Firecracker's dependencies (say `dep-name`) into a private git repository on my account (with the title `new-dep-name`)
- Replaced `dep-name`'s Cargo.toml entry with the SSH URL of the duplicate repository, using this format:
`dep-name = { git = "ssh://git@github.com/username/new-dep-name.git" }`

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -287,6 +287,16 @@ This will build and place the two Firecracker binaries at:
...
 - `build/cargo_target/${toolchain}/debug/jailer`.
 
+If you would like to test a new feature and work with
+dependencies on libraries located in private git repos, you
+can use the `--pub-key` and `--priv_key` flags to specify the
+path to your public and private ssh keys. Both of them are
+required for git authentication when fetching the repositories.
+
+```bash
+tools/devtool build --pub-key ~/.ssh/id_rsa.pub --priv-key ~/.ssh/id_rsa
```
Comment (acatangiu) [inline docs/getting-started.md:297]: Since you always need both, maybe have a single param that takes some sort of _pair_ value or at least a validation to explicitly fail or output warning when only one is provided.

File: docs/getting-started.md [hunk 2]
```diff
@@ -287,6 +287,16 @@ This will build and place the two Firecracker binaries at:
 - `build/cargo_target/${toolchain}/debug/firecracker` and
 - `build/cargo_target/${toolchain}/debug/jailer`.
 
+If you would like to test a new feature and work with
+dependencies on libraries located in private git repos, you
```
Comment (acatangiu) [inline docs/getting-started.md:291]: This only adds support for a single set of credentials. Multiple private repos with different credentials is not supported.

I think this should be fine for now, but it is worth explicitly mentioning.

**Review by acatangiu — COMMENTED** [review]
Changes look good.

Please add more detail to the PR description (maybe the commit message too).

Since this isn't tested in CI, please add a few words to the PR description around the dev-testing you did.

File: docs/getting-started.md (hunk 1 above)
Comment (luminitavoicu) [reply docs/getting-started.md:297]: Good point. I think that adding a flag that takes a pair value would work best in this case.

File: docs/getting-started.md (hunk 2 above)
Comment (luminitavoicu) [reply docs/getting-started.md:291]: Done.

**Review by gc-plp — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]
